src/core/pipelines/manifest/index.ts

```diff
@@ -82,38 +83,70 @@ export default function createManifestPipeline(
   warning$ : Subject<Error|ICustomError>
 ) : (url : string) => Observable<IFetchManifestResult> {
   const loader = createLoader<
-  IManifestLoaderArguments, Document|string
+    IManifestLoaderArguments, Document|string
   >(transport.pipelines.manifest, pipelineOptions);
 
   const parser = createParser<
-  IManifestParserArguments<Document|string>,
-  IManifestResult
+    IManifestParserArguments<Document|string>,
+    IManifestResult
   >(transport.pipelines.manifest);
 
+  const subpartPipelineOptions = objectAssign({}, pipelineOptions, {
+    loadingSubpartContent: true,
+  });
+
+  const subpartLoader = createLoader<
+    IManifestLoaderArguments, Document|string
+  >(transport.pipelines.manifest, subpartPipelineOptions);
+
   /**
    * Fetch and parse the manifest corresponding to the URL given.
    * @param {string} url - URL of the manifest
    * @returns {Observable}
    */
   return function fetchManifest(url : string) : Observable<IFetchManifestResult> {
-    return loader({ url }).pipe(
-
-      tap((arg) => {
-        if (arg.type === "error") {
-          warning$.next(arg.value);
-        }
-      }),
+    /**
+     * Request content for URL, and a given content type.
+     * Return the response.
+     * @param {string} _url
+     * @param {string} contentType
+     * @returns {Observable}
+     */
+    function loadContent<T extends string|Document>(_url: string) {
+      return loader({ url: _url }).pipe(
+        tap((arg) => {
+          if (arg.type === "error") {
+            warning$.next(arg.value);
+          }
+        }),
+        filter((arg) : arg is IPipelineLoaderResponse<T> =>
+          arg.type === "response"
+        )
+      );
+    }
 
-      filter((arg) : arg is IPipelineLoaderResponse<Document|string> =>
-        arg.type === "response"
-      ),
+    function loadSubpartContent(_url: string) {
+      return subpartLoader({ url: _url }).pipe(
+        tap((arg) => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 117. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=337&issues=AWbKt8NSO08i52vdkqIe&open=AWbKt8NSO08i52vdkqIe)

[ISSUE_KEY]: AWbKt8NSO08i52vdkqIe